federation/cmd/federation-apiserver/app/core.go

```diff
@@ -18,6 +18,14 @@ package app
 
 import (
 	"github.com/golang/glog"
+
+	// HACK to ensure that rest mapper from pkg/api is registered for groupName="".
+	// This is required because both pkg/api/install and federation/apis/core/install
+	// are installing their respective groupMeta at the same groupName.
+	// federation/apis/core/install has only a subset of resources and hence if it gets registered first, then installation of v1 API fails in pkg/master.
+	// TODO(nikhiljindal): Fix this by ensuring that pkg/api/install and federation/apis/core/install do not conflict with each other.
+	_ "k8s.io/kubernetes/pkg/api/install"
```

Comment:
The best solution is to have different instances of registered, so that clientset has its own instance and each apiserver has its own. That way we will never have pkg/api and federation/apis/core/ installing their stuff in the same registered instance.

Is that what you are doing in your PR? (it is tough to make out from the description).
If we do that, we dont need the change I talked about in my comment above.